spyder/plugins/outlineexplorer/widgets.py

```diff
@@ -266,6 +266,7 @@ def __init__(self, parent):
         self._current_editor = None
         self._languages = []
         self.is_visible = False
+        self.symbol_expanded = {}
```

Comment:
```suggestion
        self._symbols_expanded_state: dict[str, bool] = {}
```

Let's rename this attribute to be private, add typing to it and change its name to be more meaningful.